sds/src/secondary_validation/bulgarian_egn_checksum.rs

```diff
@@ -0,0 +1,71 @@
+use crate::secondary_validation::Validator;
+
+pub struct BulgarianEGNChecksum;
+
+const BULGARIAN_EGN_MULTIPLIERS: [u32; 9] = [2, 4, 8, 5, 10, 9, 7, 3, 6];
+
+impl Validator for BulgarianEGNChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        let mut chars = regex_match.chars().filter_map(|c| c.to_digit(10));
```

Comment:
here it's technically digits, not char anymore
```suggestion
        let mut digits = regex_match.chars().filter_map(|c| c.to_digit(10));
```